entry.go

```diff
@@ -85,14 +96,46 @@ func (entry *Entry) WithFields(fields Fields) *Entry {
 	return &Entry{Logger: entry.Logger, Data: data}
 }
 
+// getCaller retrieves the name of the first non-logrus calling function
+func getCaller() (method string) {
+	// Restrict the lookback to 25 frames - if it's further than that, report UNKNOWN
+	pcs := make([]uintptr, 25)
+
+	// the first non-logrus caller is at least three frames away
+	depth := runtime.Callers(3, pcs)
+	for i := 0; i < depth; i++ {
+		fullFuncName := runtime.FuncForPC(pcs[i]).Name()
+		idx := strings.LastIndex(fullFuncName, "/") + 1
+		if idx > 0 {
+			fullFuncName = fullFuncName[idx:]
+		}
+
+		matched := matchesLogrus.MatchString(fullFuncName)
+
+		// If the caller isn't part of logrus, we're done
+		if !matched {
+			if fullFuncName == "main.main" {
+				return "main"
+			} else {
+				return fullFuncName
+			}
+		}
+	}
+
+	// if we got here, we failed to find the caller's context
+	return "UNKNOWN_CALLER"
+}
+
 // This function is not declared with a pointer value because otherwise
 // race conditions will occur when using multiple goroutines
 func (entry Entry) log(level Level, msg string) {
 	var buffer *bytes.Buffer
 	entry.Time = time.Now()
 	entry.Level = level
 	entry.Message = msg
-
+	if ReportCaller() {
```

Comment:
This looks up the attribute on the std logger, which may be an incorrect assumption for this particular entry.